services/blockchain-indexer/shared/dataService/business/dpos/unlock.js

```diff
@@ -43,7 +59,46 @@ const standardizeUnlockHeight = (unlock, account, delegateAcc) => ({
 	end: calculateUnlockEndHeight(unlock, account, delegateAcc),
 });
 
+const getUnlocks = async params => {
+	const unlocks = {
+		data: {},
+		meta: {},
+	};
+
+	if (params.name) {
+		params.address = await getAddressByName(params.name);
+	}
+
+	if (params.publicKey) {
+		params.address = await getBase32AddressFromPublicKey(params.publicKey);
+	}
+
+	const response = await requestConnector('dpos_getVoter', { address: getHexAddressFromBase32(params.address) });
+	const normalizedUnlocks = response.pendingUnlocks.map(unlock => normalizeUnlock(unlock));
+
+	const accountInfo = await getIndexedAccountInfo({ address: params.address, limit: 1 }, ['name', 'publicKey']);
+	unlocks.data = {
+		address: params.address,
+		publicKey: accountInfo && accountInfo.publicKey ? accountInfo.publicKey : null,
+		name: accountInfo && accountInfo.name ? accountInfo.name : null,
+		unlocking: normalizedUnlocks,
+	};
+
```

Comment:
Could we also update the account index if the publicKey is already not indexed? We do not have to await for the operation to finish.